src/loaders/VideoLoader.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * @author mrdoob / http://mrdoob.com/
+ * @author Aaron Powell / http://lunadigital.tv
+ */
+
+import { Cache } from './Cache.js';
+import { DefaultLoadingManager } from './LoadingManager.js';
+
+
+function VideoLoader( manager ) {
+
+	this.manager = ( manager !== undefined ) ? manager : DefaultLoadingManager;
+
+}
+
+Object.assign( VideoLoader.prototype, {
+
+	crossOrigin: 'Anonymous',
+
+	load: function ( url, onLoad, onProgress, onError ) {
+
+		if ( url === undefined ) url = '';
+
+		if ( this.path !== undefined ) url = this.path + url;
+
+		url = this.manager.resolveURL( url );
+
+		var scope = this;
+
+		var cached = Cache.get( url );
+
+		if ( cached !== undefined ) {
+
+			scope.manager.itemStart( url );
+
+			setTimeout( function () {
+
+				if ( onLoad ) onLoad( cached );
+
+				scope.manager.itemEnd( url );
+
+			}, 0 );
+
+			return cached;
+
+		}
+
+		var image = document.createElementNS( 'http://www.w3.org/1999/xhtml', 'video' );
+
+		function onImageLoad() {
+
+			image.removeEventListener( 'loadeddata', onImageLoad, false );
+			image.removeEventListener( 'error', onImageError, false );
+
+			Cache.add( url, this );
+
+			if ( onLoad ) onLoad( this );
+
+			scope.manager.itemEnd( url );
+
+		}
+
+		function onImageError( event ) {
+
+			image.removeEventListener( 'loadeddata', onImageLoad, false );
+			image.removeEventListener( 'error', onImageError, false );
+
+			if ( onError ) onError( event );
+
+			scope.manager.itemEnd( url );
+			scope.manager.itemError( url );
+
+		}
+
+		image.addEventListener( 'loadeddata', onImageLoad, false );
```

Comment:
Why `loadeddata` and not `canplaythrough` or `canplay`?

Reply:
How about 'load'  ?

Reply:
'loaded' seems reasonable. The event is what did happen, not what can happen.